x-pack/plugins/alerting/server/monitoring/in_memory_metrics.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+
+export enum IN_MEMORY_METRICS {
+  RULE_EXECUTIONS = 'ruleExecutions',
+  RULE_FAILURES = 'ruleFailures',
+}
+
+export class InMemoryMetrics {
```

Comment:
Is this something that could live in the monitoringCollection plugin and be reused?

Reply:
It's a good idea, but I don't know how widespread it will be. I think we should table the idea (though I love it) for now and revisit once/if more use cases appear